presto-orc/src/main/java/com/facebook/presto/orc/OrcFileIntrospector.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc;
+
+import com.facebook.presto.orc.metadata.Footer;
+import com.facebook.presto.orc.metadata.OrcFileTail;
+import com.facebook.presto.orc.metadata.RowGroupIndex;
+import com.facebook.presto.orc.metadata.StripeInformation;
+
+import java.util.List;
+import java.util.Map;
+
+public interface OrcFileIntrospector
+{
+    default void onFileFooter(Footer fileFooter) {}
+
+    default void onFileTail(OrcFileTail fileTail) {}
+
+    default void onStripe(StripeInformation stripeInformation, Stripe stripe) {}
+
+    default void onRowGroupIndexes(StripeInformation stripe, Map<StreamId, List<RowGroupIndex>> columnIndexes) {}
```

Comment:
Should this onRowGroupIndex(Stripe, StreamId, RowGroupIndex) 

Currently the code is building the map in the reader and it might take a lot of memory. the above signature pushes the cost of building the map to the test code only when it is required.

Reply:
I think it's fine to keep it with the map.
 
From one side the introspection capabilities will only be used by tests and tools. It should not impact prod. From another side onRowGroupIndexes is called on two code paths: one for when a stripe has multiple row groups - it already works with the map, and the other path for when a stripe has a single row group.

Btw, the first code path loads bloom filters, and the second one doesn't. I wonder if it impacts any filtering use cases.

Reply:
I see, the map is already inside the ifPresent loop, so this is fine.